pkg/cloudprovider/providers/openstack/openstack_volumes.go

```diff
@@ -215,40 +215,50 @@ func (os *OpenStack) AttachDisk(instanceID, volumeID string) (string, error) {
 	if err != nil {
 		return "", err
 	}
+
 	if volume.Status != VolumeAvailableStatus {
```

Comment:
Sorry, maybe I'm confused; but if Status is not available, we'll hit this check and return an error.  In other words we'd never make it to the point of checking AttachedServerID, let alone AttachedServerID.